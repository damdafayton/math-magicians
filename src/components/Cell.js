import React from 'react';
import PropTypes from 'prop-types';

export default class Cell extends React.Component {
  render() {
    const { myClass, children } = this.props;
    return <div className={myClass}>{children}</div>;
  }
}

Cell.defaultProps = {
  myClass: '',
  children: '',
};

Cell.propTypes = {
  myClass: PropTypes.string,
  children: PropTypes.string,
};
