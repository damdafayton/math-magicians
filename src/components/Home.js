export default function Home() {
  return (
    <section className="text-start">
      <h2>Welcome to our page!</h2>
      <p>
        Mathematics is the science that deals with the logic of shape, quantity and arrangement. Math is all around us, in everything we do. It is the building block for everything in our daily lives, including mobile devices, computers, software, architecture (ancient and modern), art, money, engineering and even sports.
      </p>
      <p>
        Since the beginning of recorded history, mathematical discovery has been at the forefront of every civilized society, and math has been used by even the most primitive and earliest cultures. The need for math arose because of the increasingly complex demands from societies around the world, which required more advanced mathematical solutions, as outlined by mathematician Raymond L. Wilder in his book "Evolution of Mathematical Concepts" (Dover Publications, 2013).
      </p>
    </section>
  )
}