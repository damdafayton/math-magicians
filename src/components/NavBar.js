import { NavLink } from "react-router-dom";

export default function NavBar() {
  const navLinks = [
    {
      name: 'Home',
      to: '/'
    },
    {
      name: 'Calculator',
      to: '/calculator'
    },
    {
      name: 'Quote',
      to: '/quote'
    },
  ]

  const styles = {
    navLink: {
      listStyle: 'none',
      marginRight: '1rem'
    }
  }

  return (
    <nav className="d-flex justify-content-between align-items-center my-4 text-start w-100">
      <h1>Math Magicians</h1>
      <ul className="list-group list-group-horizontal">
        {navLinks.map((link, index) => (
          <li style={styles.navLink} >
            <NavLink className="link-dark" key={`navlink-${index}`} to={link.to} >
              {link.name}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav >
  );
}
