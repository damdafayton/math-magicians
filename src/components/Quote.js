import { useState, useLayoutEffect } from 'react';

export default function Quote() {
  const [availableHeight, setavailableHeight] = useState(0);

  useLayoutEffect(() => {
    const navElement = document.querySelector('nav');
    const navHeight = navElement && navElement.scrollHeight;
    const doc = document.scrollingElement;
    const docHeight = doc && doc.clientHeight;
    const available = docHeight - navHeight;
    if (available !== availableHeight) setavailableHeight(available);
  });

  return (
    <div className="d-flex flex-column justify-content-center" style={{ minHeight: `${availableHeight}px` }}>
      <blockquote>
        <p>
          All the records — anthropological and historical — show that counting and, ultimately,
          numeral systems as a device for counting form the inception of the mathematical element
          in all cultures,
        </p>
      </blockquote>
      <figcaption>-Raymond L. Wilder, 1968</figcaption>
    </div>
  );
}
