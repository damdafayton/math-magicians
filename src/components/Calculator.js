import React from 'react';

import Cell from './Cell';

export default class Calculator extends React.Component {
  render() {
    return (
      <div className="">
        <div className="text-white text-end">
          <Cell myClass="cell bg-dark fs-3">0</Cell>
        </div>
        <div className="row row-cols-4">
          <Cell myClass="cell bg-light">AC</Cell>
          <Cell myClass="cell bg-light">+/-</Cell>
          <Cell myClass="cell bg-light">%</Cell>
          <Cell myClass="cell bg-warning">รท</Cell>
        </div>
        <div className="row row-cols-4">
          <Cell myClass="cell bg-light">7</Cell>
          <Cell myClass="cell bg-light">8</Cell>
          <Cell myClass="cell bg-light">9</Cell>
          <Cell myClass="cell bg-warning">x</Cell>
        </div>
        <div className="row row-cols-4">
          <Cell myClass="cell bg-light">4</Cell>
          <Cell myClass="cell bg-light">5</Cell>
          <Cell myClass="cell bg-light">6</Cell>
          <Cell myClass="cell bg-warning">-</Cell>
        </div>
        <div className="row row-cols-4">
          <Cell myClass="cell bg-light">1</Cell>
          <Cell myClass="cell bg-light">2</Cell>
          <Cell myClass="cell bg-light">3</Cell>
          <Cell myClass="cell bg-warning">+</Cell>
        </div>
        <div className="row">
          <Cell myClass="cell bg-light col-6">0</Cell>
          <Cell myClass="cell bg-light col-3">.</Cell>
          <Cell myClass="cell bg-warning col-3">=</Cell>
        </div>
      </div>
    );
  }
}
