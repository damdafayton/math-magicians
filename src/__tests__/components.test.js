import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from '../components/Calculator';
import Cell from '../components/Cell';
import Home from '../components/Home';
import Quote from '../components/Quote';
import NavBar from '../components/NavBar';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  // Navlink: 'NavLink'
  NavLink: () => {
    const NavLink = 'NavLink';
    return <NavLink />;
  },
}));

const fakeFunc = () => { };

it('renders Calculator correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Cell correctly', () => {
  const tree = renderer
    .create(<Cell myClass="text-start" handleClick={fakeFunc}>fake-text</Cell>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Home correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Quote correctly', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders NavBar correctly', () => {
  const tree = renderer
    .create(<NavBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
