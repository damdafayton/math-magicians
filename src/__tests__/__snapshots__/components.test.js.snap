// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<div
  className="row flex-sm-grow-1"
>
  <p
    className="col-sm-4 text-start"
  >
    Let's do some math!
  </p>
  <section
    className="col-sm-8 px-4"
  >
    <div
      className="row text-white text-end"
    >
      <div
        className="cell bg-dark fs-3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        0
      </div>
    </div>
    <div
      className="row row-cols-4"
    >
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        AC
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        +/-
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        %
      </div>
      <div
        className="cell bg-warning"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        ÷
      </div>
    </div>
    <div
      className="row row-cols-4"
    >
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        7
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        8
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        9
      </div>
      <div
        className="cell bg-warning"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        x
      </div>
    </div>
    <div
      className="row row-cols-4"
    >
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        4
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        5
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        6
      </div>
      <div
        className="cell bg-warning"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        -
      </div>
    </div>
    <div
      className="row row-cols-4"
    >
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        1
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        2
      </div>
      <div
        className="cell bg-light"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        3
      </div>
      <div
        className="cell bg-warning"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="cell bg-light col-6"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        0
      </div>
      <div
        className="cell bg-light col-3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        .
      </div>
      <div
        className="cell bg-warning col-3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
      >
        =
      </div>
    </div>
  </section>
</div>
`;

exports[`renders Cell correctly 1`] = `
<div
  className="text-start"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  fake-text
</div>
`;

exports[`renders Home correctly 1`] = `
<section
  className="text-start"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Mathematics is the science that deals with the logic of shape, quantity and arrangement.
        Math is all around us, in everything we do.
        It is the building block for everything in our daily lives, including mobile devices, computers, software, architecture (ancient and modern), art, money, engineering and even sports.
  </p>
  <p>
    Since the beginning of recorded history, mathematical discovery has been at the forefront of every civilized society, and math has been used by even the most primitive and earliest cultures.
        The need for math arose because of the increasingly complex demands from societies around the world, which required more advanced mathematical solutions,
        as outlined by mathematician Raymond L. Wilder in his book "Evolution of Mathematical Concepts" (Dover Publications, 2013).
  </p>
</section>
`;

exports[`renders NavBar correctly 1`] = `
<nav
  className="d-flex justify-content-between align-items-center py-4 text-start w-100"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="list-group list-group-horizontal"
  >
    <li
      style={
        Object {
          "listStyle": "none",
          "marginLeft": "1rem",
        }
      }
    >
      <NavLink />
    </li>
    <li
      style={
        Object {
          "listStyle": "none",
          "marginLeft": "1rem",
        }
      }
    >
      <NavLink />
    </li>
    <li
      style={
        Object {
          "listStyle": "none",
          "marginLeft": "1rem",
        }
      }
    >
      <NavLink />
    </li>
  </ul>
</nav>
`;

exports[`renders Quote correctly 1`] = `
<div
  className="d-flex flex-column justify-content-center"
  style={
    Object {
      "minHeight": "NaNpx",
    }
  }
>
  <blockquote>
    <p>
      All the records — anthropological and historical — show that counting and, ultimately, numeral systems as a device for counting form the inception of the mathematical element in all cultures,
    </p>
  </blockquote>
  <figcaption>
    -Raymond L. Wilder, 1968
  </figcaption>
</div>
`;
